LUN-782

linkyndy opened the pull request:
Removed any extra trailing whitespace from template name; added tests.

linkyndy committed:
(LUN-782) Cleaned template name

Added a clean_fields() method on the Template model which removes any extra trailing whitespace from the template name.

linkyndy committed:
Added tests for template name cleaning

Added tests for checking whether any extra trailing whitespace is removed from the template name.

linkyndy committed:
(LUN-782) Fixed logic bug and added more tests

Validation errors are now properly shown;
Added some more tests for checking whether template name extra trailing whitespace is properly removed.

zoliszeredi commented on dbtemplates/models.py:
```diff
@@ -60,7 +60,16 @@ def populate(self, name=None):
             pass
 
+    def clean_fields(self, exclude=None):
+        """Remove extra trailing whitespace from template name (LUN-782)"""
```
You can rename this method to  clean or clean_name. This way, the method will be called automatically before save.
Also, while not explicitly stated in the ticket, you should use .strip() instead of just .rstrip()
This comment is not required, the code is sufficiently explicit.
The ticket number can be easily extracted from git, so you can drop that as well.

zoliszeredi commented on dbtemplates/models.py:
```diff
@@ -60,7 +60,16 @@ def populate(self, name=None):
+
     def save(self, *args, **kwargs):
+        # Always clean template name before save (LUN-782)
+        self.clean_fields()
```
You don't need to call clean_fields, really, its expected that the fields are cleaned by this point.

zoliszeredi commented on dbtemplates/test_cases.py:
```diff
@@ -145,3 +146,29 @@ def test_check_template_syntax(self):
+class TemplateModelNameCleanTests(TestCase):
+    """
+    Tests whether extra trailing whitespace is removed
+    from template name (LUN-782)
```
Same as above about the comments.

linkyndy committed:
(LUN-782) Review changes

Changed rstrip() with strip() and modified tests accordingly;
Removed some unnecessary comments.

linkyndy committed:
(LUN-782) More review fixes

Removed call to clean_fields() in Template.save();
Changed tests to  reflect the new behavior;

All in all, any template added via the admin form is properly cleaned, while those added via Django ORM (e.g.: Template.objects.create([...])) are saved as they are.
